Confirm to clear search

techievivek opened the pull request:
<!-- Please replace #XX below with an existing issue number. Remove the line entirely if none exist. -->
Fixes #157 

**Description**
Added the confirmation dialogue when the user clicks on the clear search button.
![ezgif com-video-to-gif](https://user-images.githubusercontent.com/35863227/75372370-72d14700-58ee-11ea-8512-b40f1ce7012a.gif)


**Other information**
Some of the design is inspired by https://github.com/creativecommons/ccsearch-browser-extension/pull/130/files

**Checklist:**
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `master` branch of the repository.
- [x] My commit messages follow [best practices][best_practices].
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

<!-- Make sure you read and understand the following attestation. -->

Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

techievivek committed:
Added user confirmation prompt while clicking on clear button

techievivek committed:
Added confirmation option while clearing the search

makkoncept reviewed (changes requested):
Please remove the lint errors and warnings.

techievivek committed:
removed extra space

techievivek commented:
@makkoncept I have updated the PR please have a look.

techievivek committed:
removed extra space

techievivek committed:
removed extra space

techievivek committed:
removed extra space

techievivek committed:
removed extra space

techievivek committed:
removed extra space

makkoncept reviewed (changes requested):
The design of the popup can be improved. You can look online for inspiration, but following are some points.
- The gradient is not visually appealing, particularly on the buttons.
- Generally, when we click anywhere outside the alert popups, it closes automatically. In this case "NO" is taken as the default response.
- Also, add a small cross icon on the top right corner which when clicked will close the popup and take "NO" as the response.
- The popup is very big, for the one-liner question.

techievivek commented:
@makkoncept Thanks for the feedback.
I will reduce the size of the popup.
Should I remove the `No` button and just have a cross icon, and yes when the user clicks on background, the popup will be closed.

makkoncept commented:
Let's keep the "NO" button. Otherwise, it can be confusing for some users.

techievivek commented:
@makkoncept Ok, then we will have ok, no and cross icon at the top-right, I will edit these changes by today and update the PR.

techievivek committed:
changed the styling of modal

techievivek committed:
Fixed linter issue

techievivek commented:
@makkoncept This is the output for the latest PR.
![ezgif com-video-to-gif](https://user-images.githubusercontent.com/35863227/75474556-2f91d980-59bd-11ea-9b35-12b450c69556.gif)

techievivek commented:
All 3 cases are implemented clicking on buttons, cross icon and background.

makkoncept commented:
@techievivek You can start implementing changing the confirmation popup to a reusable utility. I think it will best to review the PR after that.

techievivek committed:
Added a reusable modal content supporting custome modal content, confirm call, cancel call

techievivek committed:
Fixed linter issue

techievivek committed:
fixed linter issue

techievivek committed:
Fixed linter issue in basejs

techievivek committed:
Added modal btn as parameter to showmodal

techievivek committed:
Added elements as parameter to showmodal

techievivek committed:
Added elements as parameter to showModal

techievivek committed:
Added destructuring to elements

techievivek committed:
Fixed linter issue

techievivek committed:
fixed linter issue

techievivek commented:
@makkoncept I have added the modal utility, please have a look and give feedback.

makkoncept commented on src/popup/popup.html:
```diff
@@ -204,6 +204,18 @@ <h6 class="vocab heading info__label margin-bottom-medium">Share</h6>
+          <button class="modal-confirm">Yes</button>
+          <button class="modal-cancel">No</button>
+        </div>
+      </div>
```
The indentation here is very off.

makkoncept commented on src/popup/popup.html:
```diff
@@ -204,6 +204,18 @@ <h6 class="vocab heading info__label margin-bottom-medium">Share</h6>
         </div>
       </div>
     </div>
+  <div id="myModal" class="modal">
```
Remove the id `myModal`. Only use the class.

makkoncept commented on src/popup/popup.html:
```diff
@@ -204,6 +204,18 @@ <h6 class="vocab heading info__label margin-bottom-medium">Share</h6>
+  <div id="myModal" class="modal">
+        <!-- Modal content -->
+        <div class="modal-content">
+       <h3 class="popup-header">Alert</h3>
```
Giving this the class of `popup-header` can be confusing. Use `modal-header`.

makkoncept commented on src/popup/popup.html:
```diff
@@ -204,6 +204,18 @@ <h6 class="vocab heading info__label margin-bottom-medium">Share</h6>
+          <div class="popup-text" id="modal-text">
+            <p align="centre"></p>
+          </div>
+          <button class="modal-confirm">Yes</button>
```
Use class `modal-button--confirm`.

makkoncept commented on src/popup/popup.html:
```diff
@@ -204,6 +204,18 @@ <h6 class="vocab heading info__label margin-bottom-medium">Share</h6>
+       <h3 class="popup-header">Alert</h3>
+<span class="modal-close">&#10006;</span>
+          <div class="popup-text" id="modal-text">
+            <p align="centre"></p>
```
What is the use of this (`<p align="centre"></p>`)? In your implementation, this paragraph is overwritten by using`innerText` and therefore the content is not centered.

techievivek committed:
remove id and added class for various modal fields

techievivek commented:
@makkoncept I have edited the PR as per your suggestions, please have a look.

makkoncept commented on src/popup/sass/components/_confirm_popup.scss:
```diff
@@ -0,0 +1,75 @@
+  border:1px solid #ee5b32; 
+  cursor: pointer;
+  padding:4px 8px;
+}
```
- Change the name of the file to _modal.scss because that's the name this component is being referenced as everywhere.
- This file looks like just another CSS file as not a single feature of SASS has been used. We can at least nest the rules so that redundant property can be reused.

techievivek replied:
Should I make it as plain CSS file.

makkoncept replied:
No. Make use of SASS features like nested rules to refactor the stylesheet

makkoncept commented on src/popup/popup.html:
```diff
@@ -204,6 +204,17 @@ <h6 class="vocab heading info__label margin-bottom-medium">Share</h6>
+       <h3 class="modal-header">Alert</h3>
+           <span class="modal-close">&#10006;</span>
+          <div class="modal-text">
+          </div>
```
```
          <div class="modal-text">
          </div>
```
Put these in a single line.

makkoncept commented on src/popup/popup.html:
```diff
@@ -204,6 +204,17 @@ <h6 class="vocab heading info__label margin-bottom-medium">Share</h6>
         </div>
       </div>
     </div>
+  <div class="modal">
```
The indentation is still not right:
- The opening div should align with the above closing div tag.
- The closing div should align with the below opening footer tag.
- The indentation is of 2 spaces (mentioned under [editorconfig heading in INSTRUCTIONS.md](https://github.com/creativecommons/ccsearch-browser-extension/blob/master/INSTRUCTIONS.md#editorconfig)) but you have used 6 at most places, and 5 and 3 also.

techievivek replied:
Sorry for the recurrent mistake, I will keep in mind from next time.

techievivek committed:
Added proper indentation and renamed to _modal.scss

techievivek commented:
@makkoncept I have update the PR, please have a look.

techievivek committed:
Added nested rule for modal-buttons

techievivek commented:
@makkoncept I have updated the PR with nested rules for both the buttons of modal. Please have a look.

makkoncept commented on src/utils.js:
```diff
@@ -1,5 +1,26 @@
 
+export function showModal(modalText, onModalConfirm, onModalClose, elements) {
+  const {
+    modal,
```
There is no need to pass `elements` as the argument. Instead, import and use it.

makkoncept commented on src/popup/popup.js:
```diff
@@ -286,18 +286,27 @@ elements.searchIcon.addEventListener('click', () => {
+    elements.modal.style.display = 'none';
+  }
+  function onModalClose() {
+    elements.modal.style.display = 'none';
```
It would be good if you use `display-none` CSS utility defined in `_utilities.scss`.

makkoncept commented on src/popup/popup.js:
```diff
@@ -286,18 +286,27 @@ elements.searchIcon.addEventListener('click', () => {
-
+elements.modal.addEventListener('click', () => {
+  elements.modalCancel.click();
+});
```
This will close the modal even if we click on it. If the user, while trying to click the "YES" button, accidentally clicks at the side, it will be closed. This is not what we want.

makkoncept commented on src/popup/popup.js:
```diff
@@ -286,18 +286,27 @@ elements.searchIcon.addEventListener('click', () => {
+    elements.gridPrimary.setAttribute('style', 'position: relative; height: 0px;');
+    localStorage.clear();
+    restoreInitialContent('primary');
+    applyFilters();
```
Don't write this code again. Use the already defined functions for this.

techievivek replied:
I am unable to get the defined function for this, can you please tell me which function and file have this code.

makkoncept replied:
I have checked but other functions are not in the same context. I have seen the same kind of code at several places in the extension but all of them has a difference of couple lines of code. What we can do is make a utility function which is also aware of the context (like `showNotification` or `addspinner` functions).

For this PR, you can skip this.

makkoncept commented on src/popup/sass/components/_modal.scss:
```diff
@@ -0,0 +1,66 @@
+  display: none;
+  position: fixed;
+  z-index: 10;
+  padding-top: 140px;
```
All the below rules can be nested inside `.modal`rule.

techievivek replied:
I have update the PR but for this feedback I am unable to understand how to approach, also these rules are not being repeated.

makkoncept replied:
I have added the suggestion below.

techievivek committed:
removed passing to elements to showmodal and better closing of modal

techievivek committed:
Fixed linter issue

techievivek committed:
Fixed linter issue

techievivek commented:
@makkoncept I have updated the PR but for 2 suggestions I am unable to approach, Please help me so that I can proceed.

makkoncept commented on src/popup/sass/components/_modal.scss:
```diff
@@ -0,0 +1,65 @@
+  @extend .modal-button;
+  left:180px;
+  }
+}
```
```suggestion
.modal {
  position: fixed;
  z-index: 10;
  padding-top: 140px;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  overflow: auto;
  background-color: rgba(1, 0, 0, 0.4);

  &-content {
    background-color: #fefefe;
    margin: auto;
    padding: 20px;
    border: 1px solid #888;
    width: 63%;
    height: 56%;
  }

  &-close {
    float: right;
    font-size: 20px;
    font-weight: bold;
    position: relative;
    top: -50px;
    left: 13px;
    cursor: pointer;
  }

  &-text {
    padding: 20px 5px 20px 5px;
  }

  &-header {
    background-image: linear-gradient(to right, #ff4e14, #ff7d31);
    margin-top: 0;
    text-align: center;
    padding: 5px;
  }

  &-button {
    position: relative;
    display: inline-block;
    text-decoration: none;
    border: 1px solid #ee5b32;
    cursor: pointer;
    padding: 4px 8px;
    text-align: center;
    background-color: #ee5b32;
    font-size: 20px;
    color: white;
    top: 10px;
    margin: 0px 2px;
    &--cancel {
      @extend .modal-button;
      left: 60px;
    }
    &--confirm {
      @extend .modal-button;
      left: 180px;
    }
  }
}

```

makkoncept replied:
This is what I was talking about. Yes, there are not many repeated properties, but we can build from this and refactor it more. Also, this way it's easy to maintain and modify in the future. I think we should skip more refactoring of the stylesheet in this PR.

techievivek committed:
refractor as provided by Mayank

techievivek committed:
Added dark mode support for modal

techievivek commented:
@makkoncept I have added the code suggestion by you, also updated to support dark mode for modal, which I completely forgot about. Please have a look.

makkoncept commented:
Nice catch on the dark mode. Looks good. Merging.
